lib/rspec/buildkite/analytics/session.rb

```diff
@@ -134,11 +142,9 @@ def handle(_connection, data)
     def write_result(result)
       result_as_json = result.as_json
```

Comment:
I think I've commended on how the `as_json` method is mighty confusingly named. IIRC it's because we didn't want to overload to_json or to_h? It's doubly confusing because it's not returning json at all but a hash.

This makes all the places where `to_json` gets called in this PR seem odd as well. My brain is going "but this is already json, why are we calling `to_json` on it?

I think now might be a good time for a bit of a refactor/rename.